src/node_crypto.cc

```diff
@@ -3496,36 +3496,33 @@ void Sign::SignUpdate(const FunctionCallbackInfo<Value>& args) {
   sign->CheckThrow(err);
 }
 
-static int Node_SignFinal(EVPMDPointer&& mdctx, unsigned char* md,
-                          unsigned int* sig_len,
-                          const EVPKeyPointer& pkey, int padding,
-                          int pss_salt_len) {
+static bool Node_SignFinal(EVPMDPointer&& mdctx, unsigned char** md,
+                           size_t* sig_len,
+                           const EVPKeyPointer& pkey, int padding,
+                           int pss_salt_len) {
   unsigned char m[EVP_MAX_MD_SIZE];
   unsigned int m_len;
 
-  *sig_len = 0;
   if (!EVP_DigestFinal_ex(mdctx.get(), m, &m_len))
-    return 0;
+    return false;
+
+  *sig_len = static_cast<size_t>(EVP_PKEY_size(pkey.get()));
```

Comment:
`EVP_PKEY_size()` returns an int. Casting to unsigned means a negative value will wrap around.

While unlikely, there are one or two key types where this function can potentially fail (with RSA and DSA keys.)

Long story short, please `CHECK_GE(key_size, 0)` first. :-)